impala/_thrift_api.py

```diff
@@ -122,15 +126,25 @@ def __init__(self, uri_or_host, port=None, path=None, cafile=None, cert_file=Non
       self.proxy_auth = self.basic_proxy_auth_header(parsed)
     else:
       self.realhost = self.realport = self.proxy_auth = None
-    self.__auth_cookie_names = auth_cookie_names
-    self.__auth_cookie = None
-    self.__auth_cookie_expiry = None
+    # Check if 'http_cookie_names' is explicitly set as an empty value (None, [], or '').
+    if (http_cookie_names is None) or (not http_cookie_names) \
+        or (str(http_cookie_names).strip() == ""):
+      self.__http_cookie_dict = None
+    else:
+      if isinstance(http_cookie_names, six.string_types):
+        http_cookie_names = [http_cookie_names]
+      # Build a dictionary that maps cookie name to namedtuple.
+      self.__http_cookie_dict = dict()
+      for cn in http_cookie_names:
+        self.__http_cookie_dict[cn] = Cookie(cookie=None, expiry_time=None)
+    self.__is_http_cookie_set = False
```

Comment:
Could you describe the meaning of this flag in a comment? I find the its usage a bit confusing  (it is set when matching cookies are found in a response, but later it is reset if all the cookies have expired). Do we even need it?

Reply:
Renamed the variable and added the comments. I think we still need this variable when deleting cookies and re-try.